tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/security/se/SearchEngineUserReader.java

```diff
@@ -32,13 +33,15 @@ public class SearchEngineUserReader implements UserReader {
   @Override
   public Optional<UserDTO> getCurrentUserBy(final Authentication authentication) {
     final Object principal = authentication.getPrincipal();
-    if (principal instanceof User) {
-      final User user = (User) principal;
+    if (principal instanceof CamundaUser) {
+      final CamundaUser user = (CamundaUser) principal;
       return Optional.of(
           new UserDTO()
               .setUserId(user.getUserId())
               .setDisplayName(user.getDisplayName())
-              .setPermissions(rolePermissionService.getPermissions(user.getRoles()))
+              .setPermissions(
```

Comment:
Question - On this case, what is the difference between permissions and roles?

Reply:
I don't know in detail as I only refactored the code. but with my understanding about permissions' in identity I can guess that permissions are like "write:*" but roles are like "OWNER", "USER" which has a set of permissions assigned to it.